### PR TITLE
220 tooldetails

### PR DESCRIPTION
**Pull request by rishikeshDpp**

### CODE REVIEW

File: src/assets/github_icon.svg
Comment (mohitrajane) [inline src/assets/github_icon.svg:1]: please use remixicon here `<i class="ri-github-fill"></i>`

File: src/assets/presentation_icon.svg
Comment (mohitrajane) [inline src/assets/presentation_icon.svg:1]: please use remixicon `<i class="ri-slideshow-3-fill"></i>`

File: src/components/DeveloperStudio/ToolDetails.tsx [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+import PresentationIcon from "../../assets/presentation_icon.svg?react";
+import GithubIcon from "../../assets/github_icon.svg?react";
+
 function ToolDetails() {
   const { id } = useParams();
+  const {
+    data: tool,
+    isFetching: isLoadingTool,
+    error,
+    isError,
```
Comment (mohitrajane) [inline src/components/DeveloperStudio/ToolDetails.tsx:17]: remove these if not used

File: src/components/DeveloperStudio/ToolDetails.tsx [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
+  } = useShowTool(id);
 
   const navigate = useNavigate();
```
Comment (mohitrajane) [inline src/components/DeveloperStudio/ToolDetails.tsx:21]: add 
`if(isLoadingTool){
return <PageLoader />
}`
this helps user to understand the data is loading and bellow you wont have to do `tool?.` everywhere

File: src/components/DeveloperStudio/ToolDetails.tsx [hunk 3]
```diff
@@ -24,13 +33,13 @@ function ToolDetails() {
...
         <div className="tooldetail__header__details">
-          <img
-            className="tooldetail__header__details__img"
-            src={NoToolIconImg}
-            alt=""
-          />
+          <picture>
+            <source srcSet={tool?.icon[0]} type="image/webp" />
+            <source srcSet={NoToolIconImg} type="image/webp" />
+            <img src={tool?.icon[0]} className="tooldetail__header__details__img" alt="" />
```
Comment (mohitrajane) [inline src/components/DeveloperStudio/ToolDetails.tsx:39]: do we need this img tag here?

File: src/components/MySpace/Tools/NoDeveloperAccess.tsx [hunk 4]
```diff
@@ -3,20 +3,45 @@ import { Link } from "react-router-dom";
...
+          <p>Developer access request has been rejected</p>
+          <Button>
+            <Link to="/my-space/request/developer">
+              {t("profile.requestDevAccess")}
+            </Link>
+          </Button>
+        </>
+      )
+    }
+    else return (
```
Comment (mohitrajane) [inline src/components/MySpace/Tools/NoDeveloperAccess.tsx:27]: is `else return` required can't we do `return` only

File: src/components/MySpace/Tools/index.tsx [hunk 5]
```diff
@@ -41,19 +43,24 @@ function UserTools() {
...
+    if (isLoading) {
       return <Spinner size="sm" />;
     }
 
-    if (!isDeveloper) {
-      return <NoDeveloperAccess />;
+    if (!isLoading && !isDeveloper) {
+      if (!isUserRequestLoading && userRequest && userRequest?.length > 0) {
+        if (Object.keys(userRequest[userRequest?.length - 1]?.status)[0] !== "approved") {
+          return <NoDeveloperAccess access={Object.keys(userRequest[userRequest?.length - 1]?.status)[0]} />
```
Comment (mohitrajane) [inline src/components/MySpace/Tools/index.tsx:53]: why do we need Object.keys here?

**Review by mohitrajane — CHANGES_REQUESTED** [review]

File: src/assets/github_icon.svg
Comment (rishikeshDpp) [reply src/assets/github_icon.svg:1]: On it

File: src/assets/presentation_icon.svg
Comment (rishikeshDpp) [reply src/assets/presentation_icon.svg:1]: On it

File: src/components/DeveloperStudio/ToolDetails.tsx (hunk 1 above)
Comment (rishikeshDpp) [reply src/components/DeveloperStudio/ToolDetails.tsx:17]: On it, thought pratheesh might use it

File: src/components/DeveloperStudio/ToolDetails.tsx (hunk 2 above)
Comment (rishikeshDpp) [reply src/components/DeveloperStudio/ToolDetails.tsx:21]: Checking

File: src/components/MySpace/Tools/NoDeveloperAccess.tsx (hunk 4 above)
Comment (rishikeshDpp) [reply src/components/MySpace/Tools/NoDeveloperAccess.tsx:27]: True, on it

File: src/components/MySpace/Tools/index.tsx (hunk 5 above)
Comment (rishikeshDpp) [reply src/components/MySpace/Tools/index.tsx:53]: Screenshot 2024-11-29 at 12 51 33 PM

Status in response is a key with a null value, forgot to ask you about this

Comment (mohitrajane) [reply src/components/MySpace/Tools/index.tsx:53]: ah okay in future pr can you create a utility function to convert these back and forth

Comment (rishikeshDpp) [reply src/components/MySpace/Tools/index.tsx:53]: https://elna.atlassian.net/browse/ELNA-280
Added to backlog

**Pull request merged**